From 2cdab0acc99796d9e885697c6d9e50255bb3956b Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sun, 30 Aug 2015 03:21:57 -0400
Subject: [PATCH] RHEL: Fixes for libguestfs-winsupport.

In tests we cannot use guestfish for arbitrary Windows edits.
In virt-v2v helpers we must set the program name to virt-v2v.

For RHEL 9.3 and above, see this comment:
https://bugzilla.redhat.com/show_bug.cgi?id=2187961#c1
---
 convert/convert.ml                         |  1 +
 test-data/phony-guests/make-windows-img.sh |  1 +
 tests/test-block-driver.sh                 |  6 +++++-
 tests/test-in-place.sh                     |  8 +++++++-
 tests/test-virtio-win-iso.sh               |  8 +++++++-
 tests/test-windows-conversion.sh           |  8 +++++++-
 tests/test-windows-phony.sh                | 13 ++++++++++++-
 7 files changed, 40 insertions(+), 5 deletions(-)

diff --git a/convert/convert.ml b/convert/convert.ml
index ff82ad54..45d48a95 100644
--- a/convert/convert.ml
+++ b/convert/convert.ml
@@ -53,6 +53,7 @@ let rec convert input_disks options source =
 
   message (f_"Opening the source");
   let g = open_guestfs ~identifier:"v2v" () in
+  g#set_program "virt-v2v";
   g#set_memsize (g#get_memsize () * 2);
   (* Setting the number of vCPUs allows parallel mkinitrd, but make
    * sure this is not too large because each vCPU consumes guest RAM.
diff --git a/test-data/phony-guests/make-windows-img.sh b/test-data/phony-guests/make-windows-img.sh
index b52d2522..8d036ee6 100755
--- a/test-data/phony-guests/make-windows-img.sh
+++ b/test-data/phony-guests/make-windows-img.sh
@@ -49,6 +49,7 @@ fi
 
 # Create a disk image.
 guestfish <<EOF
+set-program virt-testing
 sparse $IMAGENAME-t 512M
 run
 
diff --git a/tests/test-block-driver.sh b/tests/test-block-driver.sh
index 595e969c..0b8ac6ea 100755
--- a/tests/test-block-driver.sh
+++ b/tests/test-block-driver.sh
@@ -100,7 +100,11 @@ check_driver_presence ()
 
     local virtio_dir="/Windows/Drivers/VirtIO"
 
-    guestfish --ro -a "$img" -i >$response <<-EOM
+    guestfish >$response <<-EOM
+        add-ro $img
+        set-program virt-testing
+        run
+        mount-ro /dev/sda2 /
         is-dir $virtio_dir
         is-file $virtio_dir/$drv.cat
         is-file $virtio_dir/$drv.inf
diff --git a/tests/test-in-place.sh b/tests/test-in-place.sh
index e66d2647..38d05de5 100755
--- a/tests/test-in-place.sh
+++ b/tests/test-in-place.sh
@@ -89,6 +89,12 @@ mktest ()
 :> "$script"
 :> "$expected"
 
+cat >> "$script" <<EOF
+  set-program virt-testing
+  run
+  mount /dev/sda2 /
+EOF
+
 firstboot_dir="/Program Files/Guestfs/Firstboot"
 mktest "is-dir \"$firstboot_dir\"" true
 mktest "is-file \"$firstboot_dir/firstboot.bat\"" true
@@ -101,7 +107,7 @@ for drv in netkvm vioscsi viostor; do
     done
 done
 
-guestfish --ro -a "$img" -i < "$script" > "$response"
+guestfish --ro -a "$img" < "$script" > "$response"
 diff -u "$expected" "$response"
 
 # Test the base image remained untouched
diff --git a/tests/test-virtio-win-iso.sh b/tests/test-virtio-win-iso.sh
index ccc42550..cd92db92 100755
--- a/tests/test-virtio-win-iso.sh
+++ b/tests/test-virtio-win-iso.sh
@@ -82,6 +82,12 @@ mktest ()
 :> "$script"
 :> "$expected"
 
+cat >> "$script" <<EOF
+  set-program virt-testing
+  run
+  mount /dev/sda2 /
+EOF
+
 firstboot_dir="/Program Files/Guestfs/Firstboot"
 mktest "is-dir \"$firstboot_dir\"" true
 mktest "is-file \"$firstboot_dir/firstboot.bat\"" true
@@ -94,5 +100,5 @@ for drv in netkvm vioscsi viostor; do
     done
 done
 
-guestfish --ro -a "$d/windows-sda" -i < "$script" > "$response"
+guestfish --ro -a "$d/windows-sda" < "$script" > "$response"
 diff -u "$expected" "$response"
diff --git a/tests/test-windows-conversion.sh b/tests/test-windows-conversion.sh
index bfe04904..eeddcb86 100755
--- a/tests/test-windows-conversion.sh
+++ b/tests/test-windows-conversion.sh
@@ -76,6 +76,12 @@ mktest ()
 :> "$script"
 :> "$expected"
 
+cat >> "$script" <<EOF
+  set-program virt-testing
+  run
+  mount /dev/sda2 /
+EOF
+
 firstboot_dir="/Program Files/Guestfs/Firstboot"
 mktest "is-dir \"$firstboot_dir\"" true
 mktest "is-file \"$firstboot_dir/firstboot.bat\"" true
@@ -83,7 +89,7 @@ mktest "is-dir \"$firstboot_dir/scripts\"" true
 virtio_dir="/Windows/Drivers/VirtIO"
 mktest "ls \"$virtio_dir\"" "$(cat test-windows-conversion-ls.txt)"
 
-guestfish --ro -a "$d/windows-sda" -i < "$script" > "$response"
+guestfish --ro -a "$d/windows-sda" < "$script" > "$response"
 diff -u "$expected" "$response"
 
 # We also update the Registry several times, for firstboot, and (ONLY
diff --git a/tests/test-windows-phony.sh b/tests/test-windows-phony.sh
index fa5f6a17..3f5dfe5e 100755
--- a/tests/test-windows-phony.sh
+++ b/tests/test-windows-phony.sh
@@ -71,6 +71,17 @@ mktest ()
 :> "$script"
 :> "$expected"
 
+cat >> "$script" <<EOF
+  set-program virt-testing
+  run
+  inspect-os
+  mount /dev/sda2 /
+EOF
+
+cat >> "$expected" <<EOF
+/dev/sda2
+EOF
+
 firstboot_dir="/Program Files/Guestfs/Firstboot"
 mktest "is-dir \"$firstboot_dir\"" true
 mktest "is-file \"$firstboot_dir/firstboot.bat\"" true
@@ -80,7 +91,7 @@ mktest "ls \"$virtio_dir\"" "$(cat test-phony-$guestname-ls.txt)"
 osinfo_name="${guestname%-32}"
 mktest "inspect-get-osinfo /dev/sda2" "$osinfo_name"
 
-guestfish --ro -a "$d/$guestname-sda" -i < "$script" > "$response"
+guestfish --ro -a "$d/$guestname-sda" < "$script" > "$response"
 diff -u "$expected" "$response"
 
 # host osinfo-db may be too old for win2k25
